clash-starters/clash-example-project/README.md

```diff
@@ -0,0 +1,137 @@
+
+## Cabal (Linux, MacOS)
+**The following instructions only work for Cabal >=3.0, GHC >=8.4, and Clash >=1.2.2.**
+
+First, update your cabal package database:
+
+```bash
+cabal update
+```
+
+You only have to run the update command once. After that, you can keep rebuilding your project by running the build command:
+
+```bash
+cabal build
+```
+
+To run the tests defined in `tests/`, use:
+
+```bash
+cabal run test-library
+```
+
+To compile the project to VHDL, run:
+
+```bash
+cabal run clash -- Example.Project --vhdl
+```
+
+
+## Nix (Linux, MacOS)
+To be done.
+
+## There's so many! Which should I pick?
+In general we recommend **Stack**. It offers a great balance between ease of use, flexibility, and reliability. On top of that, it's easy to install and use on Windows. Of course, it is not going to suit everyone. What follows is a comparison table between the different toolchains.
+
+|                     | Snap | Cabal     | Stack    | Nix      |
+|---------------------|------|-----------|----------|----------|
+| Windows             |      | ¹         | ✓        |          |
+| Linux               | ✓    | ✓         | ✓        | ✓        |
+| macOS               |      | ✓         | ✓        | ✓        |
+| Binary cache²       | ✓    |           |          | ✓        |
+| Ease of use         | Easy | Moderate³ | Easy     | Hard⁴    |
+| Flexibility⁵        | Low  | Moderate  | Moderate | High     |
+| Snapshots⁵          |      | ⁶         | ✓        | ✓        |
+| `ghcide` compatible |      | ✓         | ✓        | Probably |
+
+Notes in table:
+
+1. Although Cabal does run on Windows, as of the time of writing -May 2020- it doesn't offer an easy way to install itself.
+2. Binary caches store project dependencies in their binary form on some centralized system. This helps to greatly reduce build times. See [the NixOS page on binary caches](https://nixos.wiki/wiki/Binary_Cache) for more information.
+3. Ease of use is set to _moderate_ for now as it:
+   * ..does not manage GHC installations. Users are responsible for installing the right `ghc` and passing it to cabal.
+   * ..offers multiple ways of compiling a project ("old style" and "new style") which is reflected in the, scattered, cabal user documentation.
+   * ..is hard to install for Windows users
+4. Nix is notoriously hard to setup. However, many users claim that once it's setup it's a breeze to use. YMMV.
+5. TODO
+6. Cabal offers snapshots through "freeze files". Freeze files pin all (transitive) dependencies of a project to a specific version. In contrast, Stack and Nix offer snapshots of versions of Haskell packages known to work together. In other words, when using Cabal the burden of figuring out which dependency works with which is on the user.
+
+# IDE support
+We currently recommend Visual Studio Code in combination with `ghcide`. To use it, execute the following steps:
```

Comment:
You might, I still recommend `vim` :wink:

Reply:
Vim users don't need a tutorial :P. I'll add some pointers anyway!